orcid-core/src/main/java/org/orcid/core/manager/impl/OrcidProfileManagerImpl.java

```diff
@@ -423,10 +423,8 @@ private void addSourceToWorks(OrcidProfile orcidProfile, String amenderOrcid) {
     private void addSourceToWorks(OrcidWorks orcidWorks, String amenderOrcid) {
         if (orcidWorks != null && !orcidWorks.getOrcidWork().isEmpty() && amenderOrcid != null) {
             for (OrcidWork orcidWork : orcidWorks.getOrcidWork()) {
-                if (orcidWork.getSource() == null || StringUtils.isBlank(orcidWork.getSource().retrieveSourcePath())) {
```

Comment:
@nitw-shobhit Please do the following for 1.2. You'll run into unusual implicit behavior implicit behavior. 

1. Create a sample user with a self added work.

2. Post a new work work with a client.

3. Update the work just posted.

4. Inspect the sources for all works.